### PR TITLE
Extend mesh.init_cell_orientations for interval embedded in 2d

### PR DESCRIPTION
**Pull request by lrtfm**

### CODE REVIEW

File: firedrake/mesh.py [hunk 1]
```diff
@@ -2125,25 +2125,32 @@ def init_cell_orientations(self, expr):
...
         if hasattr(self.topology, '_cell_orientations'):
             raise RuntimeError("init_cell_orientations already called, did you mean to do so again?")
 
         if isinstance(expr, ufl.classes.Expr):
-            if expr.ufl_shape != (3,):
-                raise NotImplementedError('Only implemented for 3-vectors')
+            if expr.ufl_shape not in [(2,), (3,)]:
+                raise NotImplementedError('Only implemented for 2- and 3-vectors')
         else:
             raise TypeError("UFL expression expected!")
```
Comment (ksagiyam) [inline firedrake/mesh.py:2142]:
```
if not isinstance(expr, ufl.classes.Expr):
    raise TypeError("UFL expression expected!")
if expr.ufl_shape != (self.ufl_cell().geometric_dimension(), ):
    raise ValueError(f"Mismatching shapes: expr.ufl_shape ({expr.ufl_shape}) != (self.ufl_cell().geometric_dimension(), ) (({self.ufl_cell().geometric_dimension}, ))")

**Review by ksagiyam — CHANGES_REQUESTED** [review]
Thanks. Mostly looks good to me. Could you also add a test in `tests/regression/test_interpolate.py`?

File: firedrake/mesh.py [hunk 2]
```diff
@@ -2125,25 +2125,32 @@ def init_cell_orientations(self, expr):
...
+            raise ValueError(f"Mismatching shapes: expr.ufl_shape ({expr.ufl_shape}) != (self.ufl_cell().geometric_dimension(), ) (({self.ufl_cell().geometric_dimension}, ))")
+
         fs = functionspace.FunctionSpace(self, 'DG', 0)
         x = ufl.SpatialCoordinate(self)
         f = function.Function(fs)
-        f.interpolate(ufl.dot(expr, ufl.cross(ReferenceGrad(x)[:, 0], ReferenceGrad(x)[:, 1])))
+
+        if self.topological_dimension() == 1:
+            normal = ufl.as_vector((-ReferenceGrad(x)[1, 0], ReferenceGrad(x)[0, 0]))
+        else: # self.topological_dimension() == 2
```
Comment (ksagiyam) [inline firedrake/mesh.py:2150]: lint: need two spaces before `#`.

File: tests/regression/test_interpolate.py [hunk 3]
```diff
@@ -204,6 +204,21 @@ def test_cell_orientation():
...
+    m.init_cell_orientations(x)
+
+    U = VectorFunctionSpace(m, 'CG', degree=1)
+    V = VectorFunctionSpace(m, 'CG', degree=2)
+    f = project(CellNormal(m), U)
+
+    g = interpolate(f, V)
+    h = project(f, V)
+
+    assert abs(g.dat.data - h.dat.data).max() < 1e-2
```
Comment (ksagiyam) [inline tests/regression/test_interpolate.py:219]: If we are testing if interpolation is working on 1D curve, I think the following is more concise:
```
m = CircleManifoldMesh(3)
x = SpatialCoordinate(m)
m.init_cell_orientations(x)
V = VectorFunctionSpace(m, 'DG', 0)
f = interpolate(CellNormal(m), V)
assert np.allclose(f.dat.data, [[1 / 2, sqrt(3) / 2],
                                [-1, 0],
                                [1 / 2, -sqrt(3) / 2]])
```

**Review by ksagiyam — CHANGES_REQUESTED** [review]

File: tests/regression/test_interpolate.py (hunk 3 above)
Comment (lrtfm) [reply tests/regression/test_interpolate.py]: Thanks for the suggestion. Done

**Review by ksagiyam — APPROVED** [review]

**Pull request merged**